ibis/backends/duckdb/tests/test_client.py

```diff
@@ -403,3 +404,23 @@ def test_read_csv_with_types(tmp_path, input, all_varchar):
     path.write_bytes(data)
     t = con.read_csv(path, all_varchar=all_varchar, **input)
     assert t.schema()["geom"].is_geospatial()
+
+
+def test_tables_accessor_no_reference_cycle():
+    """Test that a single reference to a connection has the desired lifetime semantics."""
+    con = ibis.duckdb.connect()
+
+    before = len(gc.get_referrers(con))
+    tables = con.tables
+    after = len(gc.get_referrers(con))
+
+    assert after == before
+
+    # valid call, and there are no tables in the database
+    assert not list(tables)
+
+    del con
```

Comment:
Is there a reason this in in the `duckdb` tests instead of the general backend tests?

Reply:
Yes, because the test requires no other references to a backend in order to pass, and when `con` is used as a fixture there's more than one reference out.

Reply:
ahh, gotcha